src/pages/CustomerTablePage/CustomerTablePage.tsx

```diff
@@ -0,0 +1,296 @@
+    invoices: Invoice[],
+    customer: Customer
+  ) => {
+    setSelectedInvoices((prev) => {
+      let newSelectedInvoices;
+      if (isChecked) {
+        newSelectedInvoices = [...prev, invoice];
+      } else {
+        newSelectedInvoices = prev.filter(
+          (inv) => inv.invoiceId !== invoice.invoiceId
+        );
+      }
+
+      const someInvoicesSelected = invoices.some((inv) =>
+        newSelectedInvoices.some((selInv) => selInv.invoiceId === inv.invoiceId)
+      );
+
+      if (someInvoicesSelected && !selectedCustomers.includes(customerId)) {
+        setSelectedCustomers((prev) => [...prev, customerId]);
+        setCustomerDetails((prev) => {
+          if (!prev.some((cust) => cust.customerId === customerId)) {
+            return [...prev, customer];
+          }
+          return prev;
+        });
+      }
+
+      if (!someInvoicesSelected) {
+        setSelectedCustomers((prev) => prev.filter((id) => id !== customerId));
+        setCustomerDetails((prev) =>
+          prev.filter((cust) => cust.customerId !== customerId)
+        );
+      }
+
+      return newSelectedInvoices;
+    });
+  };
+
+  const getActionButtonTitle = () => {
+    if (selectedInvoices.length === 0) {
+      return "Request Payment / Send Reminder";
+    }
+
+    const hasOverdueInvoices = selectedInvoices.some((invoice) =>
+      isInvoiceOverdue(invoice.dueDate)
+    );
+    const hasUpcomingInvoices = selectedInvoices.some(
+      (invoice) => !isInvoiceOverdue(invoice.dueDate)
+    );
+
+    if (hasOverdueInvoices && hasUpcomingInvoices) {
+      return "Request Payment / Send Reminder";
+    } else if (hasOverdueInvoices) {
+      return "Request Payment";
+    } else {
+      return "Send Reminder";
+    }
+  };
+
+  const handleOpenModal = () => {
+    if (selectedInvoices.length > 0) {
+      setIsOpenModal(true);
+    }
+  };
+
+  const handleCloseIcon = () => {
+    setIsOpenModal(false);
+  };
+
+  const paginateCustomers = () => {
+    const indexOfLastCustomer = currentPage * customersPerPage;
+    const indexOfFirstCustomer = indexOfLastCustomer - customersPerPage;
+    return CustomerData.slice(indexOfFirstCustomer, indexOfLastCustomer);
+  };
+
+  const toggleAccordion = (customerId: number) => {
+    setExpandedCustomerId((prev) => (prev === customerId ? null : customerId));
+  };
+
+  useEffect(() => {
+    console.log("selected customers details ==> ", customerDetails);
+  }, [customerDetails]);
+
+  useEffect(() => {
+    console.log("selected invoices ==> ", selectedInvoices);
+  }, [selectedInvoices]);
+
+  useEffect(() => {
+    if (paymentData.length > 0) {
+      const newSelectedInvoices = selectedInvoices.filter(
+        (selectedInvoice) =>
+          !paymentData.some((paymentEntry) =>
+            paymentEntry.invoiceIds.includes(selectedInvoice.invoiceId)
+          )
+      );
+
+      setSelectedInvoices(newSelectedInvoices);
+
+      const newSelectedCustomers = selectedCustomers.filter((customerId) =>
```

Comment:
can be refactored?

Reply:
Refactored by creating a separate function for updated checkboxes and then calling it in useEffect.